snapcraft_legacy/_store.py

```diff
@@ -439,34 +439,45 @@ def list_keys():
         account_key["public-key-sha3-384"]
         for account_key in account_info["account_keys"]
     }
-    if keys and enabled_keys:
-        tabulated_keys = tabulate(
-            [
-                (
-                    "*" if key["sha3-384"] in enabled_keys else "-",
-                    key["name"],
-                    key["sha3-384"],
-                    "" if key["sha3-384"] in enabled_keys else "(not registered)",
-                )
-                for key in keys
-            ],
-            headers=["", "Name", "SHA3-384 fingerprint", ""],
-            tablefmt="plain",
-        )
-        print(tabulated_keys)
-    elif not keys and enabled_keys:
-        registered_keys = "\n".join([f"- {key}" for key in enabled_keys])
-        print(
-            "No keys have been created on this system. "
-            " See 'snapcraft create-key --help' to create a key.\n"
-            "The following SHA3-384 key fingerprints have been registered "
-            f"but are not available on this system:\n{registered_keys}"
-        )
+    if enabled_keys:
+        if keys:
+            tabulated_keys = tabulate(
+                [
+                    (
+                        "*" if key["sha3-384"] in enabled_keys else "-",
+                        key["name"],
+                        key["sha3-384"],
+                        "" if key["sha3-384"] in enabled_keys else "(not registered)",
+                    )
+                    for key in keys
+                ],
+                headers=["", "Name", "SHA3-384 fingerprint", ""],
+                tablefmt="plain",
+            )
+            print(tabulated_keys)
+        else:
+            registered_keys = "\n".join([f"- {key}" for key in enabled_keys])
+            print(
+                "No keys have been created on this system. "
+                " See 'snapcraft create-key --help' to create a key.\n"
+                "The following SHA3-384 key fingerprints have been registered "
+                f"but are not available on this system:\n{registered_keys}"
+            )
     else:
-        print(
-            "No keys have been registered."
-            " See 'snapcraft register-key --help' to register a key."
-        )
+        if keys:
+            tabulated_keys = tabulate(
+                [("-", key["name"], key["sha3-384"]) for key in keys],
+                headers=["", "Name", "SHA3-384 fingerprint", ""],
+                tablefmt="plain",
+            )
+            print(
+                "No keys have been registered with this account.\n"
+                "The following keys are available on this system:"
+            )
+            print(tabulated_keys)
+        else:
+            print("No keys have been registered.")
```

Comment:
it seems that this path should print:

"No keys have been created on this system. "
" See 'snapcraft create-key --help' to create a key."

as well

Reply:
Agreed! I've changed it up a bit so now it always hints about registering keys if none are registered and about creating keys if none are available on the system (effectively decoupling them except when displaying registered keys that are unavailable on the system). The constants at the top of `test_keys.py` give an example of how those might look.